hackathon/tests/test_views.py

```diff
@@ -34,3 +34,64 @@ def test_render_hackathon_list(self):
         self.assertQuerysetEqual(response.context['page_obj'],
                                  Hackathon.objects.all().order_by('-created'),
                                  transform=lambda x: x)
+
+    def test_render_hackathon_detail(self):
+        """Tests the correct rendering of the hackathon detail page,
+        including contexts."""
+
+        response = self.client.get('/hackathon/1/')
+
+        # Confirms the correct template, context items
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'hackathon/hackathon_detail.html')
+        self.assertTemplateNotUsed(response,
+                                   'hackathon/includes/enroll.html')
+        self.assertTrue(response.context['hackathon'])
+        self.assertEqual(response.context['hackathon'],
+                         Hackathon.objects.get(pk=1))
+
+        # Confirms the "enroll.html" template shows only for staff.
+        user = User.objects.get(pk=1)
```

Comment:
I would add instantiating the user in the `setUp` method.

Reply:
This was a commit and code written by @Ri-Dearg